sandbox/grommet-app/src/pages/layouts/kinds/OpsRampDetail/OpsRampDetailTable.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Anchor,
+  Box,
+  Button,
+  DataFilters,
+  DataSummary,
+  DataSort,
+  Toolbar,
+  DataSearch,
+  Data,
+  DataTable,
+  Pagination,
+  ColumnConfig,
+  View,
+} from 'grommet';
+import {
+  StatusCriticalSmall,
+  Share,
+  SettingsOption,
+  StatusGoodSmall,
+  StatusPlaceholderSmall,
+  StatusUnknownSmall,
+} from 'grommet-icons';
+import opsRamp from '../../../../mockData/opsRamp.json';
+
+import { QuickFilters } from './QuickFilters';
+
+const defaultView = {
+  search: '',
+  sort: { property: 'name', direction: 'asc' as 'asc' | 'desc' },
+  step: 10,
+  properties: {},
+  page: 1,
+};
+
+interface NodeTableProps {
+  showResultDetails: boolean;
+  setShowResultDetails: (showResultDetails: boolean) => void;
+}
+
+type Node = {
+  name: string;
+  'ip address': string;
+  make: string;
+  model: string;
+  state: string;
+};
+
+export const OpsRampDetailTable: React.FC<NodeTableProps> = ({
+  setShowResultDetails,
+}) => {
+  const [total, setTotal] = useState(0);
+  const [result, setResult] = useState<Node[]>([]);
+  const [quickFilter, setQuickFilter] = useState('');
+  const [up, setUp] = useState(0);
+  const [down, setDown] = useState(0);
+  const [view, setView] = useState<View>(defaultView);
+  const [unknown, setUnknown] = useState(0);
+  const [undefinedState, setUndefinedState] = useState(0);
+
+  useEffect(() => {
+    const nodes = opsRamp.nodes || [];
+    const upNodes = nodes.filter(nodes => nodes.state === 'up').length;
+    const downNodes = nodes.filter(nodes => nodes.state === 'down').length;
+    const unknownNodes = nodes.filter(
+      nodes => nodes.state === 'unknown',
+    ).length;
+    const undefinedNodes = nodes.filter(
+      nodes => nodes.state === 'undefined',
+    ).length;
```

Comment:
We are looping through the same array 4 times. Can we make this more efficient and only loop through once?

```suggestion
 let upNodes = 0;
    let downNodes = 0;
    let unknownNodes = 0;
    let undefinedNodes = 0;

    nodes.forEach(node => {
      switch (node.state) {
        case 'up':
          upNodes++;
          break;
        case 'down':
          downNodes++;
          break;
        case 'unknown':
          unknownNodes++;
          break;
        case 'undefined':
          undefinedNodes++;
          break;
      }
    });
```